scripts/test-tls12-point-format-extension.py

```diff
@@ -0,0 +1,340 @@
+    print("                may be specified multiple times")
+    print(" -n num         run 'num' or all(if 0) tests instead of default(all)")
+    print("                (excluding \"sanity\" tests)")
+    print(" -x probe-name  expect the probe to fail. When such probe passes despite being marked like this")
+    print("                it will be reported in the test summary and the whole script will fail.")
+    print("                May be specified multiple times.")
+    print(" -X message     expect the `message` substring in exception raised during")
+    print("                execution of preceding expected failure probe")
+    print("                usage: [-x probe-name] [-X exception], order is compulsory!")
+    print(" --help         this message")
+
+
+def main():
+    """check if app data records with zero payload are accepted by server"""
+    conversations = {}
+    host = "localhost"
+    port = 4433
+    num_limit = None
+    run_exclude = set()
+    expected_failures = {}
+    last_exp_tmp = None
+
+    argv = sys.argv[1:]
+    opts, argv = getopt.getopt(argv, "h:p:e:n:x:X:", ["help"])
+    for opt, arg in opts:
+        if opt == '-h':
+            host = arg
+        elif opt == '-p':
+            port = int(arg)
+        elif opt == '--help':
+            help_msg()
+            sys.exit(0)
+        elif opt == '-e':
+            run_exclude.add(arg)
+        elif opt == '-n':
+            num_limit = int(arg)
+        elif opt == '-x':
+            expected_failures[arg] = None
+            last_exp_tmp = str(arg)
+        elif opt == '-X':
+            if not last_exp_tmp:
+                raise ValueError("-x has to be specified before -X")
+            expected_failures[last_exp_tmp] = str(arg)
+        else:
+            raise ValueError("Unknown option: {0}".format(opt))
+
+    if argv:
+        run_only = set(argv)
+    else:
+        run_only = None
+
+    conversations = {}
+
+    # Permissive list of default ciphers, with ECDHE.
+    ciphers = [CipherSuite.TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA,
+               CipherSuite.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,
+               CipherSuite.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+               CipherSuite.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+               CipherSuite.TLS_EMPTY_RENEGOTIATION_INFO_SCSV]
+
+    # Setup the base group info; we'll duplicate this per test scenario
+    # (w.r.t. point format) desired.
+    ext = {}
+    groups = [GroupName.secp256r1,
+             GroupName.ffdhe2048]
+    ext[ExtensionType.supported_groups] = SupportedGroupsExtension()\
+        .create(groups)
+    ext[ExtensionType.signature_algorithms] = \
+        SignatureAlgorithmsExtension().create(SIG_ALL)
+    ext[ExtensionType.signature_algorithms_cert] = \
+        SignatureAlgorithmsCertExtension().create(SIG_ALL)
+    ext_missing = ext.copy()
+
+    # Extension present, no point formats supported.
```

Comment:
The below will create a malformed extension, the syntax specifies the contents as
```
ECPointFormat ec_point_format_list<1..2^8-1>
```
Which mean that an _empty_ list would be encoded as `b"\x00"`, not `b""`. The latter specifies a _missing_ list.

Reply:
Ah, d'oh. Yes, that's a good fix.